Improve documentation of Android TV Remote (#27273)
home-assistant/home-assistant.github.io: source/_integrations/androidtv_remote.markdown

You can use the `remote.turn_off`, `remote.turn_on`, `remote.toggle`,  and `remote.send_command` services from the [remote](/integrations/remote/) platform.

For a list of the most common commands you can send to the Android TV via `remote.send_command` see: [TvKeys](https://github.com/tronikos/androidtvremote2/blob/main/TvKeys.txt).

> For a full list see [here](https://github.com/tronikos/androidtvremote2/blob/main/src/androidtvremote2/remotemessage.proto#L90) (note most of the keys in the full list typically don't work).

Reviewer: We should not link to source code, this will scare end-users. Let's remove it and document what is needed.

Author: I removed it. The TvKeys.txt link above already links to the full list.

Reviewer: The point being, that link shouldn't be there either.

Author: Done.

If `activity` is specified in `remote.turn_on` it will open the specified URL in the associated app via [deep linking](https://developer.android.com/training/app-links/deep-linking).

Examples of URLs to pass as activity for some popular apps:

| App | URL |
| --- | --- |
| YouTube | `https://www.youtube.com` or `vnd.youtube://` or `vnd.youtube.launch://`
| Netflix | `https://www.netflix.com/title`
| Prime Video | `https://app.primevideo.com`
| Disney+ | `https://www.disneyplus.com`
| Plex | `plex://`

> See [this Guide/wiki](https://community.home-assistant.io/t/android-tv-remote-app-links-deep-linking-guide/567921) for how to find out what URL to use and a collection of more apps.

Reviewer: Why is that a community guide?

If it matter for the core integration, it should be part of the documentation.

Author: I don't feel the linked guide should be part of this documentation. I think it's enough to say here the integration accepts deep links. How to figure them out and having a complete list of them is outside the scope of this documentation.

Reviewer: We don't support community guides (hence the name/location).

The fact we need to point to there, indicates our main documentation is not sufficient. We need to extend the documentation in that case, not link to other community resources to resolve the problem.

../Frenck

Author: I removed the link to the guide for now.

## Services

You can use the `remote.turn_off`, `remote.turn_on`, `remote.toggle`,  and `remote.send_command` services
from the [remote](/integrations/remote/) platform.

For a list of the most common commands that you can send to the Android TV via `remote.send_command`,

> see: [TvKeys](https://github.com/tronikos/androidtvremote2/blob/main/TvKeys.txt).

Reviewer: Same comment as the previous occurance, we shouldn't be linking people into source code/repositories. The documentation should cover things like this.

Author: Done.

  entity_id: remote.living_room_tv
```

## Known issues

- Doesn't work with Fire TV devices because they are missing the [Android TV Remote Service](https://play.google.com/store/apps/details?id=com.google.android.tv.remote.service).
- If you cannot use the Google TV mobile app or the Google Home mobile app to send commands to the device, you cannot with this integration either.
- Some devices, like Xiaomi, become unavailable after they are turned off and can't be turned on with this integration.

> - Some devices experience disconnects every 15 seconds. This is resolved with rebooting the device after the initial setup of the integration.

Reviewer: ```suggestion
- Some devices experience disconnects every 15 seconds. This is resolved by rebooting the device after the initial setup of the integration.
```

- Some devices, like Xiaomi, become unavailable after they are turned off and can't be turned on with this integration.
- Some devices experience disconnects every 15 seconds. This is typically resolved by rebooting the Android TV device after the initial setup of the integration.
- If you are not able to connect to the Android TV device, or asked to pair it again and again, try force stopping the Android TV Remote Service and clearing its storage. On the Android TV device, go to settings, apps, show system apps, select Android TV Remote Service, storage, clear storage. You will have to pair again.

> - Some onscreen keyboards enabled by TV manufactures are known not to support concurrent virtual and onscreen keyboard use. This presents whenever a text field is selected such as "search" where a constant "use the keyboard on your mobile device" will show preventing you from opening the onscreen keyboard to type. This can be overcome by disabling your "3rd party" keyboard and using the default Gboard keyboard.

Reviewer: ```suggestion
- Some onscreen keyboards enabled by TV manufacturers do not support concurrent virtual and onscreen keyboard use. This presents whenever a text field is selected, such as "search" where a constant "use the keyboard on your mobile device" will show, preventing you from opening the onscreen keyboard to type. This can be overcome by disabling your "3rd party" keyboard and using the default Gboard keyboard.
```

Author: Thanks for fixing! I copied this from the last comment in https://github.com/home-assistant/core/issues/94063 and I missed the typo.
